src/wallet/wallet2.cpp

```diff
@@ -7489,34 +7482,37 @@ bool wallet2::sign_multisig_tx_from_file(const std::string &filename, std::vecto
   return sign_multisig_tx_to_file(exported_txs, filename, txids);
 }
 //----------------------------------------------------------------------------------------------------
-uint64_t wallet2::get_fee_percent(uint32_t priority, int fee_algorithm) const
+uint64_t wallet2::get_fee_percent(uint32_t priority, txtype type) const
 {
-  static constexpr std::array<std::array<uint64_t, 4>, 3> percents = {{
-    {{100, 400, 2000, 16600}},
-    {{100, 500, 2500, 100000}},
-    {{100, 500, 2500, 12500}},
-  }};
+  static constexpr std::array<uint64_t, 4> percents{{100, 500, 2500, 12500}};
 
-  if (fee_algorithm == -1)
-    fee_algorithm = get_fee_algorithm();
+  const bool blinkable = type == txtype::standard;
 
-  // 0 -> default (here, x1 till fee algorithm 2, x4 from it)
   if (priority == 0)
+  {
     priority = m_default_priority;
-  if (priority == 0)
-    priority = fee_algorithm >= 1 ? 2 : 1;
+    if ((priority == tx_priority_blink && !blinkable) || priority == 0)
+      priority = tx_priority_unimportant;
+  }
+
+  // If it's a blinkable tx then we only have two relevant priorities: unimportant, and blink.
+  if (blinkable && priority != tx_priority_unimportant)
+    priority = tx_priority_blink;
```

Comment:
On unblinkable transactions you can send the blink priority, say via the Wallet RPC or Wallet API in and it'll construct a TX with a burn. Currently this is checked only in the case of a the default/auto priority (value 0). 

We should catch that here and reject the transaction

Reply:
Done.